windowsservercore-insider/Dockerfile

```diff
@@ -1,11 +1,18 @@
 # escape=`
-FROM mcr.microsoft.com/windows/insider:10.0.18362.113
-
+FROM mcr.microsoft.com/windows/servercore/insider:10.0.19035.1
 
 RUN powershell -Command `
     Add-WindowsFeature Web-Server; `
     Invoke-WebRequest -UseBasicParsing -Uri "https://dotnetbinaries.blob.core.windows.net/servicemonitor/2.0.1.6/ServiceMonitor.exe" -OutFile "C:\ServiceMonitor.exe"
 
+RUN powershell -Command `
+    New-Item -ItemType Directory C:\LogMonitor; `
+    Invoke-WebRequest -UseBasicParsing -Uri "https://github.com/microsoft/windows-container-tools/releases/download/v1.0/LogMonitor.exe" -OutFile "C:\LogMonitor\LogMonitor.exe"; `
+    Invoke-WebRequest -UseBasicParsing -Uri "https://raw.githubusercontent.com/microsoft/iis-docker/master/windowsservercore-insider/LogMonitorConfig.json" -OutFile "C:\LogMonitor\LogMonitorConfig.json"
+
+RUN c:\windows\system32\inetsrv\appcmd.exe set config  -section:system.applicationHost/sites /"[name='Default Web Site'].logFile.logTargetW3C:"File,ETW""  /commit:apphost
+
 EXPOSE 80
 
-ENTRYPOINT ["C:\\ServiceMonitor.exe", "w3svc"]
+SHELL ["C:\\LogMonitor\\LogMonitor.exe", "powershell.exe"]
```

Comment:
Just curious why `powershell.exe` here?

Reply:
As an aside, this step will be pretty slow since the .NET Framework bits aren't NGENed in the image already. I'd skip Powershell if I can avoid it.

Reply:
Can all powershell invocation be consolidated into 1 script block of the first powershell invocation? If so, it might be more readable to have the urls in a array and have a loop that goes through the array and runs invoke-webrequest on each url. Also can we download all artifacts into 1 folder or does servicemonitor.exe have to be in the root dir?

Reply:
We can choose either cmd.exe or porwershell.exe. If using powershell is slower than using cmd.exe, I think we should use cmd.exe. I will replace with cmd.exe.

Reply:
What if we don't specify any parameters here and just let logmonitor.exe launch the entrypoint command? Does do we need a intermediate process b/w logmonitor & servicemonitor?

Reply:
That won't work. Let me explain that the SHELL ["C:\\LogMonitor\\LogMonitor.exe", "cmd.exe", "/S", "/C"] is supposed to override the default shell. In windows container, the default shell is SHELL ["cmd.exe", "/S", "/C"] and so that means we will run "C:\LogMonitor\Logmonitor.exe cmd.exe /S /C" instead of the default shell "cmd.exe /S /C". That also makes the entrypoint "c:\servicemonitor.exe" be running as the child process of LogMonitor.exe process.  If we don't override the default shell, the logmonitor.exe and the servicemonitor.exe will run separately and there is no way to pass the Ctrl signal between the two processes.

Reply:
So currently your command line for launching servicemonitor is this correct?

"C:\LogMonitor\LogMonitor.exe cmd.exe /s /c c:\servicemonitor.exe w3svc"

My question is whether cmd.exe is required?

Reply:
That's a GOOD point. Yes, it is required if we want to use SHELL. However, if we don't use the SHELL and use only ENTRYPOINT, we don't need to specify the cmd.exe because cmd.exe is the default shell anyway. Considering the fact that using the SHELL syntax makes it difficult understand, I will update to use the ENTRYPOINT only.